<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ProcessorWindow</class>
 <widget class="QMainWindow" name="ProcessorWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>691</width>
    <height>471</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ProcessorWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>671</width>
      <height>401</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Main</string>
     </attribute>
     <widget class="QPushButton" name="runButton">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>290</y>
        <width>70</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>RUN</string>
      </property>
     </widget>
     <widget class="QPushButton" name="stopButton">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>290</y>
        <width>70</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>STEP</string>
      </property>
     </widget>
     <widget class="QGraphicsView" name="monitor">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>10</y>
        <width>264</width>
        <height>264</height>
       </rect>
      </property>
     </widget>
     <widget class="QGroupBox" name="registersGrp">
      <property name="geometry">
       <rect>
        <x>300</x>
        <y>10</y>
        <width>161</width>
        <height>81</height>
       </rect>
      </property>
      <property name="title">
       <string>Registers</string>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
      <property name="checkable">
       <bool>false</bool>
      </property>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>57</width>
         <height>15</height>
        </rect>
       </property>
       <property name="text">
        <string>R1</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>57</width>
         <height>15</height>
        </rect>
       </property>
       <property name="text">
        <string>R2</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>20</y>
         <width>57</width>
         <height>15</height>
        </rect>
       </property>
       <property name="text">
        <string>A</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>40</y>
         <width>57</width>
         <height>15</height>
        </rect>
       </property>
       <property name="text">
        <string>B</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>57</width>
         <height>15</height>
        </rect>
       </property>
       <property name="text">
        <string>R3</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>60</y>
         <width>57</width>
         <height>15</height>
        </rect>
       </property>
       <property name="text">
        <string>C</string>
       </property>
      </widget>
     </widget>
     <widget class="QTextBrowser" name="txtEditor">
      <property name="geometry">
       <rect>
        <x>300</x>
        <y>110</y>
        <width>256</width>
        <height>201</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="resetButton">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>290</y>
        <width>70</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>RESET</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>Asm</string>
     </attribute>
     <widget class="QTextBrowser" name="assemblyEditor">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>50</y>
        <width>291</width>
        <height>301</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>false</bool>
      </property>
      <property name="html">
       <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'.SF NS Text'; font-size:13pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;;&lt;span style=&quot; color:#c0c0c0;&quot;&gt; &lt;/span&gt;Program&lt;span style=&quot; color:#c0c0c0;&quot;&gt; &lt;/span&gt;to&lt;span style=&quot; color:#c0c0c0;&quot;&gt; &lt;/span&gt;copy&lt;span style=&quot; color:#c0c0c0;&quot;&gt; &lt;/span&gt;and&lt;span style=&quot; color:#c0c0c0;&quot;&gt; &lt;/span&gt;determine&lt;span style=&quot; color:#c0c0c0;&quot;&gt; &lt;/span&gt;length&lt;span style=&quot; color:#c0c0c0;&quot;&gt; &lt;/span&gt;of&lt;span style=&quot; color:#c0c0c0;&quot;&gt; &lt;/span&gt;string&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;.origin&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;1000&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;start:&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;   mov&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;#msga,&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;r1&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt;   &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;mov&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;#msgb,&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;r2&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt;   &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;clr&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;r0&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;l1:&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;movb&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;(r1)+,&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;(r2)+&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt;   &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;beq&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;done&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt;   &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;inc&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;r0&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt;   &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;br&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;l1&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;done:&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;halt&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;msga:&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;.string&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;&amp;quot;A&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;string&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;whose&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;length&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;is&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;to&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;be&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;determined&amp;quot;&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;msgb:&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;.string&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;&amp;quot;Different&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;string&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;that&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;should&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;get&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;overwritten&amp;quot;&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;.end&lt;/span&gt;&lt;span style=&quot; font-family:'Monaco'; color:#c0c0c0;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Monaco';&quot;&gt;start&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
     <widget class="QTextBrowser" name="machineCodeEditor">
      <property name="geometry">
       <rect>
        <x>340</x>
        <y>50</y>
        <width>291</width>
        <height>301</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Courier New</family>
       </font>
      </property>
     </widget>
     <widget class="QPushButton" name="assemblyButton">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>10</y>
        <width>113</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>Assembly</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>691</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
